NewStarterProject/Controllers/AccessControlController.cs

```diff
@@ -0,0 +1,88 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Microsoft.Identity.Client;
+using NewStarterProject.Dtos;
+using NewStarterProject.Model;
+using NewStarterProject.Services;
+
+namespace NewStarterProject.Controllers
+{
+    [ApiController]
+    [Route("[controller]")]
+    public class AccessControlController : ControllerBase
+    {
+
+        private readonly AccessControlService _accessControlService;
+
+        public AccessControlController(StarterProjectContext context)
+        {
+            _accessControlService = new AccessControlService(context);
+        }
+
+
+        // POST: api/<AccessControlController>/ValidateAccess/5
+        [HttpPost("{id}")]
+        public async Task<ActionResult> Details(int id, [FromBody] UserDto userDto)
+        {
+            var result = await _accessControlService.ValidateAccess(id, userDto);
+
+            if (result)
+                return Accepted();
+
+            return Unauthorized();
+        }
+
+
+        // GET: api/<AccessControlController>/AllowAccessToDoor/5/1
+        [HttpGet("/AllowAccessToDoor/{doorId}/{userId}")]
+        public async Task<ActionResult> Create(int doorId, int userId)
+        {
+            var accessControlRecord = await _accessControlService.AllowAccessToDoor(doorId, userId);
```

Comment:
I really enjoy seeing what you have done here. 

In P10 we have too much logic in some of our controllers. 

Having an AccessControlService that handles the logic and returns the results is perfect and this is what we are hoping to move towards for P10. Well done!